lib/src/lints/cyclomatic_complexity/models/cyclomatic_complexity_parameters.dart

```diff
@@ -1,19 +1,25 @@
-/// Cyclomatic complexity metric limits configuration.
+import 'package:solid_lints/src/lints/cyclomatic_complexity/models/max_cyclomatic_complexity_parameters.dart';
+import 'package:solid_lints/src/models/ignored_entities_model/ignored_entities_model.dart';
+
+/// Config parameters for the `cyclomatic_complexity` rule
 class CyclomaticComplexityParameters {
-  /// Threshold cyclomatic complexity level, exceeding it triggers a warning.
-  final int maxComplexity;
+  /// `max_complexity` configuration
+  final MaxCyclomaticComplexityParameters maxCyclomaticComplexity;
```

Comment:
I do not think you need to create a separate class for extra params. Just go ahead and inherit this parameters class from the `IgnoredEntitiesModel`.

Reply:
I think it's easier to implement `fromJson`  this way. Otherwise:
```dart
class CyclomaticComplexityParams extends IgnoredEntitiesModel {
  // ...  
  factory CyclomaticComplexityParams.fromJson(Map<String, Object?> json) {
    // With more properties, I'll have to go through all of them:
    final entities = IgnoredEntitiesModel.fromJson(json).entities;

    return CyclomaticComplexityParams(
      entities: entities,
      other: 1,
    );
  }
}
```